jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/TemporaryDirectory.java

```diff
@@ -32,12 +32,13 @@
   private final Path temporaryDirectory;
 
   /**
-   * Creates a new temporary directory.
+   * Creates a new temporary directory under {@code parentDirectory}.
    *
+   * @param parentDirectory the directory to create the temporary directory within
    * @throws IOException if an I/O exception occurs
    */
-  public TemporaryDirectory() throws IOException {
-    temporaryDirectory = Files.createTempDirectory(null);
+  public TemporaryDirectory(Path parentDirectory) throws IOException {
+    temporaryDirectory = Files.createTempDirectory(parentDirectory, null);
```

Comment:
`Files.createTempDirectory` will error if `parentDirectory` does not exist. Should we be worried?

Reply:
Will delegate the creation of the parent directory to the caller. Touched up the javadoc for that.